### PR TITLE
Added utime function (set file access and modification times)

### PR DESCRIPTION
**Pull request by crystalct**

### CODE REVIEW

File: ppu/librt/utime.c [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+		struct utimbuf now_time;
+		time_t     now;
+		now = time(NULL);
+		now_time.actime = now;
+		now_time.modtime = now;
+		ret = sysLv2FsUtime(filename,(sysFSUtimbuf *)&now_time);
+	}
+	else ret = sysLv2FsUtime(filename,(sysFSUtimbuf *)times);
+	if (ret != SYS_FS_OK ) {
+		switch (ret) {
```
Comment (bucanero) [inline ppu/librt/utime.c:21]: The error conversion is already available here:
https://github.com/ps3dev/PSL1GHT/blob/master/ppu/librt/lv2errno.c

Comment (crystalct) [reply ppu/librt/utime.c]: Good!!
So i can remove utime.h

**Comment by crystalct** [comment]
I will remake a new pull request.

**Comment by miigotu** [comment]
This PR was almost right lol, just need to fix the includes I think. I can squash it when merging, so dont worry about adding 100 commits to accomplish 1 thing.

**Comment by crystalct** [comment]
@miigotu 
Something not working....
despite #include <utime.h>
function utime is not found:
`implicit declaration of function 'utime'; did you mean 'utimes'? [-Werror=implicit-function-declaration]`

**Comment by bucanero** [comment]
I think the `utime()` definition needs to be added to `utime.h`:

```c
int utime(const char *path, const struct utimbuf *times);
```

**Comment by zeldin** [comment]
Yeah, putting it in `ppu/include/utime.h` isn't right, it should go in the `sys/utime.h` installed by newlib.

**Comment by shagkur** [comment]
Shouldnt utime be declared in time.h?

Marcus Comstedt <notifications@github.com> schrieb am Sa. 4. Juli 2020 um
15:48:

**Comment by zeldin** [comment]
@shagkur No, `utime` is not a standard C function, it's a POSIX extension.

**Comment by shagkur** [comment]
True. It needs to go into sys/celllv2 or so

Marcus Comstedt <notifications@github.com> schrieb am Sa. 4. Juli 2020 um
16:20:

**Comment by zeldin** [comment]
Yes, there should be a `newlib/libc/sys/lv2/sys/utime.h`, which will then replace the stub in `newlib/libc/include/sys/utime.h` (which defines the `struct utimbuf` but no function prototypes) as `$PS3DEV/ppu/powerpc64-ps3-elf/include/sys/utime.h`

**Comment by crystalct** [comment]
As told @zeldin, prototipe is missing in `$PS3DEV/ppu/powerpc64-ps3-elf/include/sys/utime.h`, insert it manually inside it, libcurl was correctly compiled.
A little brief: 
Last polarssl/mbedtls that i have been able to compile is mbedtls-1.3.22, after polarssl stop to exist, mbedtls start to be 2.x and it needs too many network libs that there aren't in PSL1GHT (IPv6 can't be removed with config, too).
Last libcurl with polarssl support is curl-7.65.0, compiled OK.
Open ssl (1.1.1g), compiled OK, but i haven't been able to convince libcurl config to see open ssl installation :-(

**Comment by bucanero** [comment]
I've just submitted a PR to the newlib repo to add the missing definition: https://github.com/ps3dev/newlib-1.20.0-PS3/pull/3

As I understand, this would also require to update the newlib patch in `ps3toolchain`.

**Pull request closed without merging**